query-node/schemas/membership.graphql

```diff
@@ -1,9 +1,21 @@
+type AvatarObject @variant {
+  "The avatar data object"
+  avatarObject: DataObject!
+}
+
+type AvatarUri @variant {
+  "The avatar URL"
+  avatarUri: String!
+}
+
+union Avatar = AvatarObject | AvatarUri
+
 type MemberMetadata @entity {
   "Member's name"
   name: String
 
   "Avatar data object"
-  avatar: DataObject
+  avatar: Avatar
```

Comment:
For some reason the processor fails when no avatar is passed:

> ERROR: {} name: Error, message: QueryFailedError: null value in column "avatar" violates not-null constraint, stack: Error: QueryFailedError: null value in column "avatar" violates not-null constraint

The query is 

```
ERROR: Stopping the proccessor due to errors: {
  "message": "null value in column \"avatar\" violates not-null constraint",
  "length": 291,
  "name": "QueryFailedError",
  "severity": "ERROR",
  "code": "23502",
  "detail": "Failing row contains (TQmS4ou_u, 2021-12-08 13:27:54.004, ZJqpET1FvN, 2021-12-08 13:27:54.004, null, null, null, 1, alice, null, null).",
  "schema": "public",
  "table": "member_metadata",
  "column": "avatar",
  "file": "execMain.c",
  "line": "1948",
  "routine": "ExecConstraints",
  "query": "INSERT INTO \"member_metadata\"(\"id\", \"created_at\", \"created_by_id\", \"updated_at\", \"updated_by_id\", \"deleted_at\", \"deleted_by_id\", \"version\", \"name\", \"avatar\", \"about\") VALUES ($1, $2, $3, $4, DEFAULT, DEFAULT, DEFAULT, $5, $6, DEFAULT, DEFAULT) RETURNING \"created_at\", \"updated_at\", \"version\"",
  "parameters": [
    "TQmS4ou_u",
    "2021-12-08T13:27:54.004Z",
    "ZJqpET1FvN",
    "2021-12-08T13:27:54.004Z",
    1,
    "alice"
  ]
}
```

Reply:
If this is because of variant type then maybe just "Empty" would be OK :thinking:

Reply:
That's strange `null` should be valid.
I fixed it by passing an empty string in `avatar.avatarUri`